lib/internal/logger/serializers.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+const { isNativeError } = require('internal/util/types');
+
+/**
+ * Serializes an Error object
+ * @param {Error} error
+ * @returns {object}
+ */
+function serializeErr(error) {
+  if (!isNativeError(error)) {
+    return error;
+  }
+
+  const obj = {
+    type: error.constructor.name,
+    message: error.message,
+    stack: error.stack,
+  };
+
+  // Include additional error properties
+  for (const key in error) {
+    if (obj[key] === undefined) {
+      obj[key] = error[key];
+    }
+  }
+
+  // Handle error code if present
+  if (error.code !== undefined) {
+    obj.code = error.code;
+  }
+
+  // Handle error cause recursively
+  if (error.cause !== undefined) {
+    obj.cause = typeof error.cause === 'object' && error.cause !== null ?
+      serializeErr(error.cause) :
+      error.cause;
+  }
+
+  return obj;
+}
+
+/**
+ * Serializes HTTP request object
+ * @param {object} req - HTTP request
```

Comment:
Can you extend the jsdoc here? Something like this, but with better indentation:

```suggestion
 * @param {{method: string, url: string, headers: string, socket: undefined | {remoteAddress: string, remotePort: string}} req - HTTP request
```